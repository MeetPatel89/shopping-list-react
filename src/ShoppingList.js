import React from 'react';
import ShoppingItem from './ShoppingItem';

export default class ShoppingList extends React.Component {

    static defaultProps = {
        items: []
    }
    
    render() {
        const shoppingItems = this.props.items.map((item, i) =>
        <ShoppingItem
          key={i}
          
          item={item}
          onDeleteItem={this.props.onDeleteItem}
          onCheckItem={this.props.onCheckItem}
        />)
        
        

        return (
            <ul>
                {shoppingItems}
            </ul>
        )
    }
}