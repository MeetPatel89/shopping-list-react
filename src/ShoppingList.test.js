import React from 'react';
import ReactDOM from 'react-dom';
import ShoppingList from './ShoppingList';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'; 

describe('ShoppingList Componenet', () => {

    test('Renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<ShoppingList/>, div);

        ReactDOM.unmountComponentAtNode(div);
    })

    test('renders empty given no items', () => {
        const wrapper = shallow(<ShoppingList/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('renders ShoppingItems when supplied with an items prop', () => {
        const testItems = [
            {name: 'test-item1', checked: false},
            {name: 'test-item2', checked: true},
            {name: 'test-items3', checked: true}
        ]
        const wrapper = shallow(<ShoppingList items={testItems}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})