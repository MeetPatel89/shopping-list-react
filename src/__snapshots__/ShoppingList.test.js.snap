// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingList Componenet renders ShoppingItems when supplied with an items prop 1`] = `
<ul>
  <ShoppingItem
    item={
      Object {
        "checked": false,
        "name": "test-item1",
      }
    }
    key="0"
  />
  <ShoppingItem
    item={
      Object {
        "checked": true,
        "name": "test-item2",
      }
    }
    key="1"
  />
  <ShoppingItem
    item={
      Object {
        "checked": true,
        "name": "test-items3",
      }
    }
    key="2"
  />
</ul>
`;

exports[`ShoppingList Componenet renders empty given no items 1`] = `<ul />`;
